game-app/game-core/src/test/java/games/strategy/triplea/delegate/VictoryTest.java

```diff
@@ -175,6 +174,194 @@ void testMotorizedNoBlitzBlitzedTerritory() {
     moveDelegate.end();
   }
 
+  @Nested
+  final class AttackingFromContestedTerritory {
+    final List<Unit> infantryUnit = infantry.create(1, italians);
+    final List<Unit> tankUnit = armour.create(1, italians);
+    final List<Unit> tankAndInfantry = List.of(tankUnit.get(0), infantryUnit.get(0));
+    final Route route = gameData.getMap().getRoute(angloEgypt, transJordan, it -> true);
+
+    @BeforeEach
+    void setUp() {
+      // Ensure that both territories we're testing are owned by an enemy.
+      assertThat(angloEgypt.getOwner().isAtWar(italians), is(true));
+      assertThat(transJordan.getOwner().isAtWar(italians), is(true));
+      // Add some enemy units to both of them.
+      gameData.performChange(ChangeFactory.addUnits(angloEgypt, armour.create(1, british)));
+      gameData.performChange(ChangeFactory.addUnits(transJordan, armour.create(1, british)));
+      // Ensure that infantry can't blitz and tank can, since we want to test both cases.
+      assertThat(Matches.unitCanBlitz().test(infantryUnit.get(0)), is(false));
+      assertThat(Matches.unitCanBlitz().test(tankUnit.get(0)), is(true));
+      // Add them to angloEgypt.
+      gameData.performChange(ChangeFactory.addUnits(angloEgypt, tankAndInfantry));
+
+      advanceToStep(testBridge, "CombatMove");
+      moveDelegate.setDelegateBridgeAndPlayer(testBridge);
+      moveDelegate.start();
+    }
+
+    @Test
```

Comment:
Consider adding javadocs to '@Test' to describe the test setup and what we are testing. Those comments describing the test I've personally found to be useful pretty soon after to go back and understand more easily what is being tested.

Reply:
Done!